### PR TITLE
Aidan/stop pod cli

### PR DESCRIPTION
**Pull request by AidanAbd**

### CODE REVIEW

File: latch_cli/main.py [hunk 1]
```diff
@@ -41,10 +41,12 @@ def main():
     latest_ver = parse_version(get_latest_package_version())
     if local_ver < latest_ver:
         click.secho(
-            textwrap.dedent(f"""
+            textwrap.dedent(
```
Comment (ayushkamat) [inline latch_cli/main.py:44]: update ur version of black to `23.7.0` and add the `--preview` flag so we don't have reformatting happening

File: latch_cli/main.py [hunk 2]
```diff
@@ -601,6 +603,42 @@ def get_executions():
...
+        except Exception:
+            click.secho(
+                "Unable to read `/root/.latch/id` -- pass as command line argument",
+                fg="red",
+            )
+            return
+
+    from latch_cli.services.stop_pod import stop_pod
+
+    stop_pod(pod_id)
```
Comment (ayushkamat) [inline latch_cli/main.py:639]:
```suggestion
def stop_pod(pod_id: Optional[int] = None):
    """Stops a pod given a pod_id or the pod from which the command is run"""
    crash_handler.message = "Unable to stop pod"

    from latch_cli.services.stop_pod import stop_pod

    if pod_id is None:
        id_path = Path("/root/.latch/id")

        try:
            pod_id = int(id_path.read_text().strip("\n"))
        except Exception as e:
            if isinstance(e, FileNotFoundError):
                err_str = f"Pod ID not found at {id_path}"
            elif isinstance(e, ValueError):
                err_str = f"Could not parse Pod ID at {id_path}"
            else:
                err_str = f"Error reading Pod ID from {id_path}"

            click.secho(
                f"{err_str} -- please provide a Pod ID as a command line argument.",
                fg="red",
            )
            return

    stop_pod(pod_id)
```

File: latch_cli/services/stop_pod.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+def stop_pod(pod_id: int) -> None:
+    """Stops a pod given a pod_id"""
+    res = tinyrequests.post(
+        urljoin(NUCLEUS_URL, "/pods/stop"),
+        headers={"Authorization": get_auth_header()},
+        json={"pod_id": pod_id},
+    )
+
+    if res.status_code == 200:
+        click.secho(f"Pod with id `{pod_id}` Stopped.", fg="green")
```
Comment (ayushkamat) [inline latch_cli/services/stop_pod.py:19]:
```suggestion
        click.secho(f"Pod with ID `{pod_id}` stopped.", fg="green")
```

or alternatively

```suggestion
        click.secho(f"Pod `{pod_id}` stopped.", fg="green")
```

File: latch_cli/services/stop_pod.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        return
+
+    if res.status_code == 403 or res.status_code == 404:
+        click.secho("Pod does not exist or permission denied.", fg="red")
+        return
```
Comment (ayushkamat) [inline latch_cli/services/stop_pod.py:24]: why not just separate the two

File: latch_cli/services/stop_pod.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    if res.status_code == 500:
+        click.secho(
+            f"Internal error while stopping pod `{pod_id}`. Please try again and"
```
Comment (ayushkamat) [inline latch_cli/services/stop_pod.py:28]:
```suggestion
            f"Internal error while stopping Pod `{pod_id}`. Please try again."
```

File: latch_cli/services/stop_pod.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            " contact `support@latch.bio` if the issue persists.",
```
Comment (ayushkamat) [inline latch_cli/services/stop_pod.py:29]:
```suggestion
            " Contact `support@latch.bio` if the issue persists.",
```

File: latch_cli/services/stop_pod.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            fg="red",
+        )
+        return
+
+    raise ValueError(f"failed to stop pod with id `{pod_id}` with code. Server responded with `{res.status_code}` status code.")
```
Comment (ayushkamat) [inline latch_cli/services/stop_pod.py:34]: del

File: latch_cli/services/stop_pod.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+    if res.status_code == 200:
+        click.secho(f"Pod with id `{pod_id}` Stopped.", fg="green")
+        return
+
+    if res.status_code == 403 or res.status_code == 404:
+        click.secho("Pod does not exist or permission denied.", fg="red")
+        return
+
+    if res.status_code == 500:
```
Comment (ayushkamat) [inline latch_cli/services/stop_pod.py:26]:
```suggestion
    if res.status_code != 200:
```

**Review by ayushkamat — CHANGES_REQUESTED** [review]

File: latch_cli/services/stop_pod.py (hunk 4 above)
Comment (AidanAbd) [reply latch_cli/services/stop_pod.py:24]: these should always be grouped -- this endpoint is stupid and does not.

**Pull request merged**